portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,122 @@
 
 package com.google.sps.servlets;
 
+import com.google.sps.data.Comment;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** 
+ * Servlet with a GET handler that loads a list of comments from Datastore and 
+ * a POST handler that stores a new comment in Datastore.
+ */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final Logger LOGGER = Logger.getLogger(DataServlet.class.getName());
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int maxComments = getMaxCommentsToReturn(request);
+
+    // Query up to maxComments comment entities from Datastore.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(maxComments));
+
+    // Construct a list of comments from the queried entities.
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results) {
+      long id = entity.getKey().getId();
+      String author = (String) entity.getProperty("author");
+      String commentText = (String) entity.getProperty("commentText");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(id, author, commentText, timestamp);
+      comments.add(comment);
+    }
+
+    // Convert the list of comments to JSON.
+    String json = convertToJson(comments);
+
+    // Send the JSON as the response.
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println(json);
+  }
+
+  /** 
+   * Returns the maximum number of comments selected by the user, or the default of 10 if the number was invalid.
+   */
+  private int getMaxCommentsToReturn(HttpServletRequest request) {
+    // Get the input from the form.
+    String maxCommentsString = request.getParameter("max-comments");
+
+    final int DEFAULT_MAX_COMMENTS = 10;
```

Comment:
I would probably even make this static so that it could be shared across all instances:

`private static final int DEFAULT_MAX_COMMENTS = 10;`